packages/terra-table/src/TableSection.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import SectionHeader from './TableSectionHeader';
+
+const propTypes = {
+  /**
+   * The children list items passed to the component.
+   */
+  children: PropTypes.node,
+  /**
+   * Number of columns the section header must span
+   */
+  colSpan: PropTypes.number.isRequired,
+  /**
+   * Whether or not the section is collapsed.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Whether or not the section can be collapsed.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  /**
+   * The associated metaData to be provided in the onSelect callback.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Function callback passthrough for the ref of the section li.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Title text to be placed within the section header.
+   */
+  title: PropTypes.string,
+};
+
+const defaultProps = {
+  children: [],
+  isCollapsed: false,
+  isCollapsible: false,
+  level: 1,
+  title: '',
+};
+
+const TableSection = ({
+  children,
+  isCollapsed,
+  isCollapsible,
+  ...customProps
+}) => {
+  let sectionItems;
+  if (!isCollapsible || !isCollapsed) {
+    sectionItems = children;
+  }
+
+  return (
+    <React.Fragment>
+      <SectionHeader {...customProps} isCollapsible={isCollapsible} isCollapsed={isCollapsed} />
```

Comment:
Should a section be able to be collapsed if is not collapsible? i.e. should this be:
```
isCollapsed={isCollapsible  && isCollapsed}
```